pyccel/parser/semantic.py

```diff
@@ -2497,7 +2497,7 @@ def _visit_FunctionDef(self, expr, **settings):
                         if isinstance(a, ValuedArgument):
 
                             # optional argument only if the value is None
-                            if isinstance(a.value, Nil):
+                            if not a.value:
```

Comment:
This needs to be the right type. (e.g. The argument is not optional if the value is `False`)

Reply:
This is true in several of the places that you have changed.
I think to work out what you can change you can follow the following guidelines:
- If `Nil` is tested but not `None` then the type is important, don't change the test
- If the test is `if X or if isinstance(X, Nil)` then you can definitely change the test
- If the test is `if (X is None) or if isinstance(X, Nil)` then the type may be important. You need to think about whether `X` can have a value other than `None` or `Nil` which could be evaluated as `False`

Reply:
It may be that you can't change many of them, but I still think it is useful for `Nil` to have a `__bool__` method anyway